core/comparison.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+
+#include "scipp/core/comparison.h"
+
+namespace scipp::core {
+
+/**
+ * Tests if the unit, values (and variances where appropriate) of two Variables
+ * are within an absolute tolerance.
+ */
+bool is_approx(Variable a, const Variable &b, const double tol) {
```

Comment:
```suggestion
bool is_approx(const VariableConstProxy &a, const VariableConstProxy &b, const double tol) {
```

Reply:
`transform_in_place` needs a mutable `Variable` as far as I can tell.

Reply:
That may be, but the interface of a comparison operator should never allow for modification of the operands.